### DISCUSSION TITLE
Will it download original RAW files but with metadata I've added in Apple Photos?

### DISCUSSION THREAD

**Discussion by bbx-git** [icloud-photos-downloader/icloud_photos_downloader#727] [Q&A]
As in title.

Cheers

**Comment by AndreyNikiforov** [comment]
`icloudpd` can download RAW files the last time I checked. Do you know how metadata is stored for your photo? If metadata is stored inside the file (e.g. EXIF), then it will be in your local copy as well.

**Reply by bbx-git** [reply]
I don't think it's stored in a file.
When in Apple Photos you select the option "Export originals" it will export photos without metadata.

That's a bummer.

**Reply by AndreyNikiforov** [reply]
Do you see this extra metadata in iCloud UI? Can you download that metadata separately? I am trying to understand if that metadata is somehow exposed and we can add support for it at some point in time (at the end `icloudpd` is emulating web access to iCloud, so if data is available in UI there is hope).

**Reply by bbx-git** [reply]
Just checked on iCloud.com and it does show all the metadata.
There is things like camera/lens info, and a map with location.

I don't see any option to export metadata.

I also wanted to try icloudpd program on my Mac but didn't manage to install it.
I've installed it via pip3 and tried to add PATH to .zprofile but never got it to work.
Would be nice if the tutorial showed exactly how to get it to work.

**Reply by AndreyNikiforov** [reply]
All metadata that are added to photo automatically (camera, dates, location etc) are available in EXIF in HEIC format. Don't know about RAW, but suspect there is high chance metadata is embedded in the file, in which case you'll get all that in your local copy.

Original question was about metadata that YOU ADDED to photos. I have no experience with RAW format, so you have to try and see.

`icloudpd` on mac has issues (see #668 #700), which are expected to be fixed in next release.

**Reply by bbx-git** [reply]
Yes I've added for example location in Apple Photos.
It doesn't matter if it's RAW or HEIC or JPG, if you export as original file then it won't have the metadata added by yourself.

**Reply by bbx-git** [reply]
Location and date are most important for me, captions would be nice too.

**Reply by AndreyNikiforov** [reply]
I am missing something. When I take an image on iDevice, date/time & location are added to the image automatically. When image is automatically uploaded to iCloud from my iDevice, all metadata is preserved. When I download that image with `icloudpd`, local copy has all metadata. That is true for HEIC/JPG, I suspect it is true for RAW, but I don't know.

If I take any photo and add date/time/location/etc to EXIF and then upload to iCloud though web UI, the image will have metadata preserved. When it is downloaded by `icloudpd` all metadata is still inside the image.

How are you taking your RAW images? At which point do you update metadata? Where do see metadata disappearing during `icloudpd` usage?

**Reply by bbx-git** [reply]
I am using a dedicated camera, and some of my photos are scanned.

If you add location or captions manually on iOS or Mac version of Photos, and then try to export them using Photos as unmoddified, it will not contain things like location.

**Reply by AndreyNikiforov** [reply]
If you have dedicated camera, I suggest the following steps to identify the problem area:
- add metadata to photos on your computer
- upload to iCloud using browser
- validate that metdata is respected by iCloud
- download using `icloudpd`
- validate that metadata preserved in your local copy

I don't know behavior of the Photos on iOS/Mac and I limit troubleshooting to `icloudpd`. For issues with Photos app, you have to reach to Apple...

**Reply by bbx-git** [reply]
Would you be able to test it yourself? I can't download it on Mac and just had an operation and feeling like dead.
If yes, then please just go to iCloud or any Apple device and add a random location or a date to a photo, and see if your app downloaded original with modified data?

Thanks a lot.
